webview/src/plots/components/customPlots/CustomPlots.tsx

```diff
@@ -82,15 +83,15 @@ export const CustomPlots: React.FC<CustomPlotsProps> = ({ plotsIds }) => {
       onDragLeave={() => setOnSection(false)}
       onDragOver={handleDragOver}
       onDrop={handleDropAtTheEnd}
+      ref={gridRef}
     >
-      <Grid
-        setOrder={setPlotsIdsOrder}
+      <CustomPlotsGrid
+        gridRef={gridRef}
```

Comment:
Why does this need to be passed down? Can't the observer be initialized in here?

Reply:
discussed IRL.

tl;dr - ref must be passed down or the grid will appear to have 0 dimensions. The alternative is to use `useLayoutEffect`.

Reply:
I just tried it, and it works correctly without changing anything else. https://github.com/iterative/vscode-dvc/pull/5111